packages/sanity/src/core/store/_legacy/grants/documentPairPermissions.ts

```diff
@@ -300,8 +300,10 @@ export function useDocumentPairPermissions({
 
   const serverActionsEnabled = useMemo(() => {
     // If it's explicitly disabled, we'll just return a stream that emits `false`
-    return workspace.serverActions?.enabled === false ? of(false) : fetchFeatureToggle(client)
-  }, [client, workspace.serverActions?.enabled])
+    return workspace.__internal_serverDocumentActions?.enabled === false
```

Comment:
Does this not need to updated like this?

```ts
return typeof workspace.__internal_serverDocumentActions?.enabled === 'boolean' ? of(workspace.__internal_serverDocumentActions?.enabled as boolean)
```

(Currently I think this would only respect a negative override).

Reply:
You're right - thanks for catching!